### PR TITLE
Fixed racy test

### PR DESCRIPTION
**Pull request by gregw**
Fixed racy test from #6563 

Signed-off-by: Greg Wilkins <gregw@webtide.com>

### CODE REVIEW

**Review by olamy — APPROVED** [review]

File: jetty-server/src/test/java/org/eclipse/jetty/server/HttpOutputTest.java [hunk 1]
```diff
@@ -767,16 +767,23 @@ public void setNext(Interceptor interceptor)
     @Test
     public void testEmptyArray() throws Exception
     {
-        AtomicBoolean committed = new AtomicBoolean();
+        FuturePromise<Boolean> committed = new FuturePromise<>();
```
Comment (lachlan-roberts) [inline jetty-server/src/test/java/org/eclipse/jetty/server/HttpOutputTest.java:770]: If you're using `FuturePromise` now, maybe you should add a timeout for the `get()` call.

Comment (gregw) [reply jetty-server/src/test/java/org/eclipse/jetty/server/HttpOutputTest.java:770]: oops I missed this comment.  I'm planning for success and the test will pass!        If we break something, then a test that hangs forever will get just as much attention than one that times out.

Comment (lachlan-roberts) [reply jetty-server/src/test/java/org/eclipse/jetty/server/HttpOutputTest.java:770]: Yeah but its not ideal, sometimes its hard to find where it timed out and why. Much better to just fail after a few seconds.

Comment (gregw) [reply jetty-server/src/test/java/org/eclipse/jetty/server/HttpOutputTest.java:770]: I've fixed this in the 10 PR which has not yet been merged.

**Pull request merged**